packages/product-json-to-csv/src/map-product-data.js

```diff
@@ -74,92 +102,258 @@ export default class ProductMapping {
+      variant,
+      (acc: Object, value: any, property: string): Object => {
+        acc[property] = this._mapVariantProperty(property, value)
+        return acc
+      },
+      {}
+    )
+
+    // complete missing attributes
+    if (productType && productType.attributes)
+      productType.attributes.forEach((attribute: Object) => {
+        const mappedAttributes = mappedVariant.attributes || {}
+        if (isNil(mappedAttributes[attribute.name]))
+          mappedAttributes[attribute.name] = ''
+      })
+
+    return mappedVariant
+  }
+
+  _mapVariantProperty(property: string, value: any): any {
+    switch (property) {
+      case 'attributes':
+        return this._mapAttributes(value)
+      case 'prices':
+        return this._mapPricesToString(value)
+      case 'images':
+        return this._mapImagesToString(value)
+      case 'availability':
+        return value.availableQuantity
+      case 'id':
+      case 'sku':
+      case 'key':
+        return value
+      default:
+        return undefined
+    }
+  }
+
+  _mapPricesToString(prices: Array<Price>): string {
+    return prices
+      .map((price: Price) => ProductMapping._mapPriceToString(price))
+      .join(this.multiValDel)
+  }
+
+  static _mapPriceToString(price: Price): string {
+    // Full price:
+    // 'country-currencyCode centAmount|discounted.centAmount customerGroup.name#channel.key$validFrom~validUntil'
+
+    return oneLineTrim`
+      ${price.country ? `${price.country}-` : ''}
+      ${price.value.currencyCode} ${price.value.centAmount}
+      ${price.discounted ? `|${price.discounted.value.centAmount}` : ''}
+      ${
+        price.customerGroup && price.customerGroup.name
+          ? ` ${price.customerGroup.name}`
+          : ''
+      }
+      ${price.channel && price.channel.key ? `#${price.channel.key}` : ''}
+      ${price.validFrom ? `$${price.validFrom}` : ''}
+      ${price.validUntil ? `~${price.validUntil}` : ''}
+    `
+  }
+
+  _mapLtextAttribute(name: string, value: Object): Object {
+    // create an object with value based on selected language
+    const mappedAttribute = {
+      [name]: value[this.lang],
+    }
+    // create object with translations indexed with keys like: attributeName.en
+    const labels = flatten({
+      [name]: value, // value contains object with localized labels
+    })
+
+    // add labels in format "attribName.en: labelEnValue" to all attributes
+    return { ...mappedAttribute, ...labels }
+  }
+
+  _mapLabelsToAllAttributes(name: string, labels: Array<Object>): Object {
+    const mappedValues: Object = {}
+    const languages: Array<string> = labels
+      .map(Object.keys) // returns array of arrays with language keys
+      // flatten subarrays to one big array
+      .reduce((all, langKeys) => {
+        all.push(...langKeys)
+        return all
+      }, [])
+      // filter out duplicates
+      .filter((value, index, self) => self.indexOf(value) === index)
```

Comment:
With `Set` it will be cloning the whole array to `Set` while removing duplicate values and then it will again transform the unique values to a new array - so it will be unnecessarily cloning array two times which will downgrade performance a little bit. I am not sure what is the optimal way here - better readability or performance?